src/ogma/features/nodeGrouping.ts

```diff
@@ -42,30 +43,20 @@ export class NodeGroupingTransformation {
           if (this._isRuleNotApplicableToNode(node)) {
             return undefined;
           } else {
-            const propertyValue = node.getData([
-              'properties',
-              this.groupRule?.groupingOptions.propertyKey ?? ''
-            ]);
-            // if the property value is of type conflict or invalid value we use the original value
-            const originalValue =
-              typeof propertyValue === 'object'
-                ? (propertyValue as ConflictValue).original
-                : propertyValue;
+            const propertyValue = this._findGroupingPropertyValue(node);
             // groupRule is defined if not we returned undefined
             // node with same value will be part of the same group
-            return `${this.groupRule?.groupingOptions.itemType}-${originalValue}`
+            return `${this.groupRule?.groupingOptions.itemType}-${propertyValue}`
               .toLowerCase()
               .trim();
           }
         },
         nodeGenerator: (nodes) => {
           return {
             data: {
```

Comment:
the size property was not used so I removed it. 
I added nodeGroupId which represents the id of the virtual node. It will be saved in the viz object and used to save information about the node group (For now we are just saving if the node is layoutable or not)

Here is the [tech specs ](https://www.notion.so/linkurious/User-want-to-pin-a-node-group-3972a4c27fe14b70b6443a0a69e43344?pvs=4#2ecad44dc07449968d57c93144694428) maybe it will be helpful

Reply:
You can retrieve it from the virtual node using:  myNode.getData('nodeGroupId')